x/stake/keeper.go

```diff
@@ -355,83 +351,71 @@ func (k Keeper) clearAccUpdateValidators(ctx sdk.Context) {
 //_____________________________________________________________________
 
 // load a delegator bond
-func (k Keeper) GetDelegatorBond(ctx sdk.Context,
-	delegatorAddr, candidateAddr sdk.Address) (bond DelegatorBond, found bool) {
+func (k Keeper) GetDelegation(ctx sdk.Context,
+	delegatorAddr, validatorAddr sdk.Address) (bond Delegation, found bool) {
 
 	store := ctx.KVStore(k.storeKey)
-	delegatorBytes := store.Get(GetDelegatorBondKey(delegatorAddr, candidateAddr, k.cdc))
+	delegatorBytes := store.Get(GetDelegationKey(delegatorAddr, validatorAddr, k.cdc))
 	if delegatorBytes == nil {
 		return bond, false
 	}
 
-	err := k.cdc.UnmarshalBinary(delegatorBytes, &bond)
-	if err != nil {
-		panic(err)
-	}
+	k.cdc.MustUnmarshalBinary(delegatorBytes, &bond)
 	return bond, true
 }
 
 // load all bonds
-func (k Keeper) getBonds(ctx sdk.Context, maxRetrieve int16) (bonds []DelegatorBond) {
+func (k Keeper) getBonds(ctx sdk.Context, maxRetrieve int16) (bonds []Delegation) {
 	store := ctx.KVStore(k.storeKey)
-	iterator := store.Iterator(subspace(DelegatorBondKeyPrefix))
+	iterator := store.SubspaceIterator(DelegationKey)
 
-	bonds = make([]DelegatorBond, maxRetrieve)
+	bonds = make([]Delegation, maxRetrieve)
 	i := 0
 	for ; ; i++ {
 		if !iterator.Valid() || i > int(maxRetrieve-1) {
 			iterator.Close()
 			break
 		}
 		bondBytes := iterator.Value()
-		var bond DelegatorBond
-		err := k.cdc.UnmarshalBinary(bondBytes, &bond)
-		if err != nil {
-			panic(err)
-		}
+		var bond Delegation
+		k.cdc.MustUnmarshalBinary(bondBytes, &bond)
 		bonds[i] = bond
 		iterator.Next()
 	}
 	return bonds[:i] // trim
```

Comment:
Is this more or less efficient than just using `append` and not allocating the whole array?